### DISCUSSION TITLE
Graph classification where nodes belong to multiple graphs

### DISCUSSION THREAD

**Discussion by evantianzg** [pyg-team/pytorch_geometric#9088] [Q&A]
Hi everyone,

I am using GNNs to perform graph classification tasks. I first used the public TUDatasets for graph classification, and it works well. Then I tried to conduct experiments with my own datasets. However, I found that there is a "graph_indicator" file in the public TUDatasets (like MUTAG). After carefully reading the description and content of this "graph_indicator" file, I suddenly realized the difference between my dataset and the public dataset: **Each node on public datasets belongs to a single graph, while in my dataset, some nodes belong to multiple graphs.** Therefore, I cannot create a similar "graph_indicator" file as in the public datasets.

**So my question is: how should I do for graph classification where nodes belong to multiple graphs?**

Any help or suggestions are appreciated.

**Comment by rusty1s** [comment]
In this case, it is best to duplicate the node for each graph.

**Reply by evantianzg** [reply]
I'm not sure if I get you completely.

If there are 2 graphs and there are 3 nodes for the first and 4 nodes for the second. Suppose the 2 graphs have a common node (so there are only 6 nodes in total). In this case, the "graph_indicator" file should be 
`[1,1,1,2,2,2,2]`.

There are actually 6 nodes in total, but there are 7 elements in the "graph_indicator" file because the common node is duplicated. Am I right?

**Reply by rusty1s** [reply]
Yes, but your graph should also have 7 nodes (not just the graph indicator file).

**Reply by evantianzg** [reply]
Definitely. My remaining question is described in Q&A #9072
